decode_test.go

```diff
@@ -995,29 +977,21 @@ func (errReader) Read([]byte) (int, error) {
 
 func TestDecoderReadError(t *testing.T) {
 	err := yaml.NewDecoder(errReader{}).Decode(&struct{}{})
-	if err == nil || !strings.Contains(err.Error(), `yaml: input error: some read error`) {
-		t.Fatalf("Decode() returned %v, want error containing %q", err, `yaml: input error: some read error`)
-	}
+	testutil.AssertErrorMatches(t, err, `yaml: input error: some read error`)
 }
 
 func TestUnmarshalNaN(t *testing.T) {
 	value := map[string]interface{}{}
 	err := yaml.Unmarshal([]byte("notanum: .NaN"), &value)
-	if err != nil {
-		t.Fatalf("Unmarshal() returned error: %v", err)
-	}
-	if !math.IsNaN(value["notanum"].(float64)) {
-		t.Fatalf("Unmarshal() returned %v, want NaN", value["notanum"])
-	}
+	testutil.AssertNoError(t, err)
+	testutil.AssertEqual(t, math.IsNaN(value["notanum"].(float64)), true)
```

Comment:
Here having an AssertTrue would be better

Reply:
done